### PR TITLE
Wells: SDK foundation, layout

### PR DESCRIPTION
**Pull request by 0xalecks**
- extracts TokenValue, Token, and some other classes into a stand alone new package: @beanstalk/sdk-core
- adds new sdk-wells package, with barebones structure. builds successfully
- examples for core and wells sdks, confirming compile/build/etc steps work

### CODE REVIEW

File: projects/examples/package.json [hunk 1]
```diff
@@ -3,14 +3,16 @@
   "main": "src/index.ts",
   "dependencies": {
     "@beanstalk/sdk": "workspace:*",
+    "@beanstalk/sdk-core": "workspace:*",
```
Comment (marshall-bean) [inline projects/examples/package.json:6]: Minor thing, but to me "core" implies "foundation" when really these are just common classes, so I'd suggest "sdk-common" as the package name instead.

File: projects/sdk-core/.gitignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+*.log
+.DS_Store
+node_modules
+dist
+cache
+.env
```
Comment (marshall-bean) [inline projects/sdk-core/.gitignore:6]: We need to ignore `src/constants/generated/**/*` and `src/constants/generated-gql/**/*` also don't we? The `generate` command in sdk-core would leave artifacts there.

File: projects/sdk-core/src/constants/abi/ERC20.json [hunk 3]
```diff
@@ -0,0 +1,117 @@
+[
```
Comment (marshall-bean) [inline projects/sdk-core/src/constants/abi/ERC20.json:1]: Should these and other src/constants be committed?

File: projects/examples/package.json [hunk 4]
```diff
@@ -3,14 +3,16 @@
...
+    "@beanstalk/wells": "workspace:*",
```
Comment (marshall-bean) [inline projects/examples/package.json:7]: Adding this here but applies to `jest.config.ts` - should we add these new projects there? Along with top-level sdk-core and sdk-wells package.json commands (e.g. sdk-core:test).

**Review by marshall-bean — CHANGES_REQUESTED** [review]
A few small things.

File: projects/examples/package.json (hunk 1 above)
Comment (0xalecks) [reply projects/examples/package.json:6]: this is a convention used by other web3 projects, like uniswap, so went with it

File: projects/examples/package.json (hunk 4 above)
Comment (0xalecks) [reply projects/examples/package.json:7]: yeah, we can once we add tests

File: projects/sdk-core/src/constants/abi/ERC20.json (hunk 3 above)
Comment (0xalecks) [reply projects/sdk-core/src/constants/abi/ERC20.json:1]: yes, we can commit the abis, these are pretty static

**Review by marshall-bean — APPROVED** [review]

**Pull request merged**